hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -40,13 +41,41 @@
 from hypothesis.internal.cathetus import cathetus
 from hypothesis.internal.renaming import renamed_arguments
 from hypothesis.utils.conventions import infer, not_set
-from hypothesis.internal.reflection import proxies, required_args
+from hypothesis.internal.reflection import proxies, required_args, \
+    define_function_signature
 from hypothesis.internal.validation import check_type, try_convert, \
-    check_strategy, check_valid_size, check_valid_bound, \
-    check_valid_sizes, check_valid_integer, check_valid_interval, \
-    check_valid_magnitude
+    check_valid_size, check_valid_bound, check_valid_sizes, \
+    check_valid_integer, check_valid_interval, check_valid_magnitude
+from hypothesis.searchstrategy.lazy import LazyStrategy
```

Comment:
This is adding a *lot* of stuff to the namespace, and even if it's private API that's going to be annoying for some people or tools (and, of course, human readers!).

Why not `import hypothesis.searchstrategy as strat`, then refer to the classes by their qualified names e.g. `strat.misc.BoolStrategy`?  IMO this would be much more readable!

Reply:
I don't believe that works. Until they've been explicitly imported, modules in packages are not actually visible as attributes of a package:

```
david@DESKTOP-1PB01VL:~/scratch/packaging$ mkdir a
david@DESKTOP-1PB01VL:~/scratch/packaging$ touch a/__init__.py
david@DESKTOP-1PB01VL:~/scratch/packaging$ touch a/b.py
david@DESKTOP-1PB01VL:~/scratch/packaging$ python
Python 3.6.0 (default, Dec 24 2016, 09:12:41)
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import a
>>> a.b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'a' has no attribute 'b'
>>> from a import b
>>> a.b
<module 'a.b' from '/home/david/scratch/packaging/a/b.py'>
```

I could add `import hypothesis.searchstrategy.misc as misc` or something but that's not much of a namespace reduction.

I think the correct solution here is to split up this file properly and make hypothesis.strategies a package that just re-exports stuff from its modules, but that's a larger change that I wanted to go into here.

Reply:
Fair enough!

I'd try importing them all into `hypothesis.searchstrategy.__init__` and using that as the namespace, but happy for you to leave it as-is for this PR.

Reply:
Yeah, that would work, but I don't really see that the namespace pollution in strategies is a major problem and I'd rather not band-aid it when it gets fixed by the proper fix automatically